scipy/stats/_bws_test.py

```diff
@@ -0,0 +1,230 @@
+           statistic - A survey." Statistical Papers, vol. 46, pp. 1-30, Springer 
+           Verlag, 2005.
+    .. [2] Shan, G., Ma, C., Hutson, A. D., & Wilding, G. E. (2013). "Some tests 
+           for detecting trends based on the modified Baumgartner–Weiß–Schindler 
+           statistics." Computational Statistics & Data Analysis, 57(1), 246-261.
+    .. [3] Michael P. Fay and Michael A. Proschan. "Wilcoxon-Mann-Whitney
+           or t-test? On assumptions for hypothesis tests and multiple \
+           interpretations of decision rules." Statistics surveys, Vol. 4, pp.
+           1-39, 2010. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2857732/
+
+    Examples
+    --------
+    We follow the example of table 3 in [1]_: Fourteen children were divided randomly into two groups, 
+    an experimental group and a control group. For each child the number of correct identifications 
+    among 18 new examples was observed.
+
+    >>> control_group_ranks = np.array([1, 2, 3, 4, 6, 7, 8])
+    >>> experimental_group_ranks = np.array([5, 9, 10, 11, 12, 13, 14])
+
+    We use the Baumgartner-Weiss-Schindler test to assess whether there is a 
+    statistically significant difference in the methods of training preschool children.
+    The null hypothesis is that the distribution of first method is the same as 
+    the distribution of second one. We decide that a confidence level of 95% is 
+    required to reject the null hypothesis in favor of the alternative that 
+    the distributions are different.
+    Since the number of samples is very small and there are no ties in the
+    data, we can compare the observed test statistic against the *exact*
+    distribution of the test statistic under the null hypothesis.
+
+    >>> from scipy.stats import bws_test
+    >>> B, p = bws_test(control_group_ranks, experimental_group_ranks, method="exact")
+    >>> print(B)
+    5.132167152575317
+
+    This agrees with :math:`B = 5.132` reported in [1]_. The *p*-value produced
+    by `bws_test` agrees with :math:`p = 0.0029` reported in [1]_.
+
+    >>> print(p)
+    0.002838
+
+    It normally takes a long time to calculate the exact *p*-value when the passing array
+    is large or contains a lot of ties, so we can approximate it using a random permutation test.
+    Here we use `method="asymptotic"` to perform a random permutation test with 10000 samples.
+    Noticeably, *p* value varies depending on the number of samples, but it normally maintains 
+    the conclusion unchanged.
+
+    >>> from scipy.stats import bws_test
+    >>> B, p = bws_test(control_group_ranks, experimental_group_ranks, method="asymptotic", numsamples=10000)
+    >>> print(p)
+    0.0038
+
+    Under this assumption, the *p*-value would be low enough to reject the
+    null hypothesis in favor of the alternative.
+    '''
+    rs = np.argsort(np.argsort(np.concatenate([x, y])))
+
+    if alternative == "greater":
+        pass  # TODO: implement
```

Comment:
Is this a temporary `pass` or is it going to be finished as such?

Reply:
I would implement it later according to https://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC4809145&blobtype=pdf

Reply:
One-sided has been implemented as well @ilayn